coq-HOAS.elpi

```diff
@@ -14,10 +14,10 @@
 % Command and tactic invocation (coq_elpi_vernacular.ml)
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
-% Entry point for commands. Eg. "Elpi mycommand foo 3 (f x)." becomes
-%   main [str "foo", int 3, trm (app[f,x])]
+% Entry point for commands. Eg. "#[att=true] Elpi mycommand foo 3 (f x)." becomes
+%   main [str "foo", int 3, trm (app[f,x])] [pr "att" (leaf "true")]
 % The encoding of terms is described below.
-pred main i:list argument.
+pred main i:list argument, i:list attribute.
```

Comment:
Excuse my ignorance, but I am not sure I understand... 
- First in `coq-builtin.elpi` the add-hoc constructor is used (not `pr`).
- Then d I get it right that your examples from https://github.com/LPCIC/coq-elpi/pull/109#issuecomment-573070631 get translated in the following way
  - `#[foo]` -> `[attribute "foo" (node [])]`
  - `#[foo="x"]` -> `[attribute "foo" (leaf "x")]` BTW, can it only be a string? If so why the double quotes `"`? I thought ot would be possible to do `#[foo=1]` or `#[foo=(fun x => x)]`... And is `#[foo=bar]` correct and represented by `[attribute "foo" (leaf "bar")]`?
  - `#[foo(bar)]`-> `[attribute "foo" (node [attribute "bar" (node [])]]`
  - `#[foo,baz]` -> `[attribute "foo" (node []), attribute "baz" (node [])]`

EDIT: after looking more intensely, I read `parse-attributes`, which if I understand correctly typechecks and interpret the attributes, which are then made available via clauses. Are `attribute` and `attribute-value` reflecting exactly what Coq's API gives, while `parse-attributes` and the typechecking are @gares add-value using elpi?